src/Core/View.js

```diff
@@ -394,18 +396,60 @@ View.prototype.getLayers = function getLayers(filter) {
  * @param {Function} frameRequester.update - update will be called at each
  * MainLoop update with the time delta between last update, or 0 if the
  * MainLoop has just been relaunched.
+ * @param {String} when - decide when the frameRequester.update function should
+ * be called during the update cycle.
  */
-View.prototype.addFrameRequester = function addFrameRequester(frameRequester) {
-    this._frameRequesters.push(frameRequester);
+View.prototype.addFrameRequester = function addFrameRequester(frameRequester, when = View.AFTER_CAMERA_UPDATE) {
+    if (typeof frameRequester.update !== 'function' && typeof frameRequester !== 'function') {
+        throw new Error('frameRequester must be a function or have a function \'update\'');
```

Comment:
Now that frameRequesters can be notified several time, what about requiring objects to have named method `onBeforeRender`, `onAfterRender` instead? This could simplify client code if they need callbacks on several hook points and want a single FrameRequester to handle them. And maybe require to specify the hook point only if frameRequester is a function?

Or maybe allow all those 3 mechanisms? So these 3 statements would be accepted:

```js
// adding a simple function
view.addFrameRequester(View.UPDATE_START, function(dt) { /* ... */ });
// adding a simple object with update fn
view.addFrameRequester(View.UPDATE_START, { update: () => { /* ... */ } });
// adding an object with some `on...` methods. We don't specify hook points in this case
view.addFrameRequester({
    onUpdateStart: /* ... */,
    onBeforeCameraUpdate: /* ... */,
    /* ... */
});
```
Maybe it's overkill I don't know. WDYT? At least n°1 and either 2 or 3 should stay supported though.

Reply:
> This could simplify client code if they need callbacks on several hook points and want a single FrameRequester to handle them.

Well they can always do something like this, which is rather simple IMHO:
```js
function frameRequester(dt) { /* ... */}
view.addFrameRequester(View.UPDATE_START, frameRequester);
view.addFrameRequester(View.UPDATE_END, frameRequester);
```
But I concur that your example can be prettier: I'm not fond of how those constants look. So we would have:
- `UPDATE_START` -> `onUpdateStart`
- `BEFORE_CAMERA_UPDATE` -> `onBeforeCameraUpdate`
- `AFTER_CAMERA_UPDATE` -> `onAfterCameraUpdate`
- `BEFORE_LAYER_UPDATE` -> `onBeforeLayerUpdate`
- `AFTER_LAYER_UPDATE` -> `onAfterLayerUpdate`
- `BEFORE_RENDER` -> `onBeforeRender`
- `AFTER_RENDER` -> `onAfterRender`
- `UPDATE_END` -> `onUpdateEnd`

I'm for one solution though, either 1+2 or 3. Any opinion of someone else on the matter ?

Reply:
Solution 3 is nicer if you want to work with an object, but how is this going to work if you want to register a function?

Other random ideas: 
```js
view.onUpdateStart.addFrameRequester(...)
view.exec(myfunction).on('View.UPDATE_START')
```

Reply:
Do you find bothering to only be able to write `view.addFrameRequester({ onUpdateStart: /* ... */ });` for one function ?

From your first idea, we could also have:
```js
view.onUpdateStart.add(...);
view.onUpdateStart.remove(...);
view.onUpdateStart.exec(...);
```

Reply:
I don't have a strong opinion about this so choose whatever suits you best :)

Reply:
> Well they can always do something like this, which is rather simple IMHO:
>```js
>function frameRequester(dt) { /* ... */}
>view.addFrameRequester(View.UPDATE_START, frameRequester);
>view.addFrameRequester(View.UPDATE_END, frameRequester);
>```

Not in the current state. If we want this to work, we need to pass the name of the current step as a parameter to the frame requester when we call it: you want to be able to know what the current step is *from inside* your frame requester. It's possible, but that would mean each frame requester that wants to be notified only for one step must start with `if (currentStep === myStep)`. A bit cumbersome if you ask me.

> Solution 3 is nicer if you want to work with an object, but how is this going to work if you want to register a function?

Solution 3 would only be supported for objects. Solution 1 and/or 2 for both objets and functions.  We can have all 3 if we want to. Not sure I understood your question actually :-)

> I'm for one solution though, either 1+2 or 3. Any opinion of someone else on the matter ?

Actually, maybe the safest solution for today would be to do 1/ only (aka `view.addFrameRequester(hookName, fn)`), because it works in all cases, it is the most js idiomatic and the simplest. So I'd say we could even drop 2/ (objects with `update` fn) for now (there would be a couple of controls to amend, but that will be very simple). Adding 2/ or 3/ later would be easy anyway, if we feel the need.

I'm not fond of the other possibilities you guys proposed.

WDYT?

Reply:
Good to me to keep only 1, I don't know what the original thought behind `update` was. I think we can all agree on 1 ? I just need to invert `fn` and `hookName` and that'll be good. We can always come back later if someone find a better solution.

Reply:
>  I don't know what the original thought behind update was.

I did that :-) My original work on FrameRequesters was to move away from having listeners everywhere (like in GlobeControls for instance) OR adding `requestAnimationFrame` in random part of the code. You know that already, but the end goal is to provide a mechanism that offers better performances and strong guarantees (be called at each frame after or before a definite step) for client code than events. 

The first application for it was controls (see every controls except GlobeControls), and as GlobeControls does have an `update` method (that was the only control at the time), I started from that. Later @peppsac added the ability to use simple functions. We still need to retrofit GlobeControls to use or be a frame requester though.

That's the story! It's quite a recent one though, so we can still experiment with this concept.

Reply:
I see, I think this may be too specific to Controls then, so better drop it for now. I'm changing all of this then !